tableExport.js

```diff
@@ -75,7 +100,13 @@ THE SOFTWARE.*/
 						console.log(tdData);
 					}
 					var base64data = "base64," + $.base64.encode(tdData);
-					window.open('data:application/'+defaults.type+';filename=exportData;' + base64data);
+					// Old method.
+					// window.open('data:application/'+defaults.type+';filename=exportData;' + base64data);
+					
+					
+					extension = (defaults.type == 'csv')?'.csv':'.txt';
```

Comment:
Here, and in all the hunks that follow, I've replaced the old windows.open() with a call to the openDownloadFileDialog().